src/main/webapp/app/exercises/programming/hestia/testwise-coverage-report/testwise-coverage-report.component.ts

```diff
@@ -0,0 +1,75 @@
+import { Component, Input, OnInit } from '@angular/core';
+import { CoverageReport } from 'app/entities/hestia/coverage-report.model';
+import { CoverageFileReport } from 'app/entities/hestia/coverage-file-report.model';
+
+@Component({
+    selector: 'jhi-testwise-coverage-report',
+    templateUrl: './testwise-coverage-report.component.html',
+})
+export class TestwiseCoverageReportComponent implements OnInit {
+    @Input()
+    report: CoverageReport;
+
+    @Input()
+    fileContentByPath: Map<string, string>;
+
+    displayedTestCaseNames: Map<string, boolean>;
+
+    fileReportByFileName: Map<string, CoverageFileReport>;
+
+    constructor() {}
+
+    ngOnInit(): void {
+        // initially display covered lines for all test cases
+        this.displayedTestCaseNames = new Map<string, boolean>();
+        // retrieve all test cases
+        const testCases = new Set(
+            this.report?.fileReports
+                ?.flatMap((report) => report)
+                .flatMap((fileReport) => fileReport.testwiseCoverageEntries)
+                .map((entry) => entry!.testCase),
+        );
+        testCases.forEach((testCase) => this.displayedTestCaseNames.set(testCase!.testName!, true));
+
+        this.setReportsByFileName();
+    }
+
+    changeReportsBySelectedTestCases(testCaseName: string): void {
+        const selected = this.displayedTestCaseNames.get(testCaseName) ?? false;
```

Comment:
This is not needed. You're inverting ``selected`` in the next line using ``!`` anyways which implicitly converts to a boolean, so it's enough if this is just any falsy value (aka ``undefined`` in this case)
```suggestion
        const selected = this.displayedTestCaseNames.get(testCaseName);
```